bot/cogs/geo_cog/streetviewrandomizer/street_view_random.py

```diff
@@ -0,0 +1,249 @@
+                loops += 1
+
+            coord = fai.coord
+            country_df = fai.country_df
+            attempts = fai.attempts
+            elapsed_time_ms = fai.total_elapsed_time_ms
+            total_attempts += attempts
+            total_elapsed_time_ms += elapsed_time_ms
+
+            country_iso3 = country_df["ISO3"].values[0]
+            country_name = country_df["NAME"].values[0]
+            gdf.loc[gdf["ISO3"] == country_iso3, "IMAGES"] += 1
+
+            logging.log(1, f"\n> Image found in {country_iso3} ({country_name}) | "
+                           f"lon: {coord.lon}, lat: {coord.lat} | attempts: {attempts} "
+                           f"| total elapsed time: {elapsed_time_ms / 1000:.2f}s\n")
+            total_elapsed_time_ms = total_elapsed_time_ms + int(await self.save_images(
+                coord, args['size'], args['headings'], args['pitches'], args['fovs']
+            ))
+            logging.log(1, f"\nImage saved!\n Total attempts: {total_attempts} Average number of attempts per "
+                           f"sampling: {total_attempts / args['samples']:.2f} "
+                           f"\nTotal elapsed time: {total_elapsed_time_ms / 1000:.2f}s Average elapsed time per "
+                           f"sampling: "
+                           f"{(total_elapsed_time_ms / 1000) / args['samples']:.2f}s")
+
+        size = args['size']
+
+        return (('https://maps.googleapis.com/maps/api/streetview?'
+                + f'size={size}&'
+                + f'location={coord.lat},{coord.lon}&'
+                + f"heading={args['headings']}&"
+                + f"pitch={args['pitches']}&"
+                + f"fov={args['fovs']}&"
+                + f"key={bot_secrets.secrets.geocode_key}"),
+                coord.lat,
+                coord.lon,
+                country_iso3,
+                new_country_selections_if_error)
+
+    @staticmethod
+    def compute_area(gdf: gpd.GeoDataFrame) -> gpd.GeoDataFrame:
+        gdf = gdf.eval("AREA = geometry.to_crs('esri:54009').area")
+        gdf = gdf.sort_values(by="AREA", ascending=False)
+
+        gdf = gdf.eval("AREA_PERCENTAGE = AREA / AREA.sum()")
+        gdf = gdf.sort_values(by="AREA_PERCENTAGE", ascending=False)
+        gdf = gdf.reset_index(drop=True)
+
+        return gdf
+
+    # Find a random point in the specified country up to 50 times or until street view is found.
+    async def find_available_image(self, gdf: gpd.GeoDataFrame, radius_m: int) -> ImageData:
+        """
+        Get an image from Google Street View Static API.
+        :param gdf: Geo-dataframe for a single target country.
+        :param radius_m: Grab the nearest streetview within a radius.
+        :return: ImageData dataclass.
+        """
+        coord: Coordinate
+        country_df = None
+        attempts = 0
+        total_elapsed_time_ms = 0
+        image_found = False
+
+        logging.log(1, str(country_df) + "looking in")
+        while not image_found:
```

Comment:
Might be more readable if this was switched to a `do...while` loop instead.

Reply:
Do while loops as they exist in C don't exist in python. The closest thing I see to a "do...while loop" online is replacing the while condition with "while true". Would you like this instead?

Reply:
Yeah, basically:
```py
while True:
    # ...
    if condition:
        break
```